lib/libxdp/libxdp.c

```diff
@@ -3223,6 +3223,43 @@ static int xdp_multiprog__unpin(struct xdp_multiprog *mp)
 	return err;
 }
 
+static void xdp_detach_links(uint32_t ifindex) {
+	struct bpf_link_info link_info;
+	u_int32_t link_info_len = sizeof(link_info), id;
+	int err, fd;
+
+	while (true) {
+		err = bpf_link_get_next_id(id, &id);
+		if (err) {
+			if (errno == ENOENT)
+				break;
+			pr_debug("can't get next link: %s (id %d)", strerror(errno), id);
+			break;
+		}
+
+		fd = bpf_link_get_fd_by_id(id);
+		if (fd < 0) {
+			pr_debug("can't get link by id (%u): %s", id, strerror(errno));
+			continue;
+		}
+		memset(&link_info, 0, sizeof(link_info));
+		err = bpf_obj_get_info_by_fd(fd, &link_info, &link_info_len);
+		if (err) {
+			pr_debug("can't get link info for %u: %s", id, strerror(errno));
+			close(fd);
+			continue;
+		}
+		if (link_info.type == BPF_LINK_TYPE_XDP && link_info.xdp.ifindex == ifindex) {
+			err = bpf_link_detach(fd);
+			if (err) {
+				pr_debug("can't detach link %u: %s", id, strerror(errno));
```

Comment:
At this point we actually found a link to detach, so if it fails we should be printing at a higher severity than debug (`pr_warn()` would be suitable I think). A `pr_debug()` would be useful either *before* attempting the detach, or on success, though...

Reply:
changed to warn and added debug before attempting to detach